waftools/detections/compiler_swift.py

```diff
@@ -19,50 +21,162 @@ def __add_swift_flags(ctx):
+                    swift_libraries[lib_type] = swift_library.abspath()
+                    break
+            else:
+                break
+
+    #check if library paths were found
+    ctx.start_msg('Checking for dynamic Swift Library')
+    if 'SWIFT_LIB_DYNAMIC' in swift_libraries:
+        ctx.end_msg(swift_libraries['SWIFT_LIB_DYNAMIC'])
+    else:
+        ctx.end_msg(False)
+
+    ctx.start_msg('Checking for static Swift Library')
+    if 'SWIFT_LIB_STATIC' in swift_libraries:
+        ctx.end_msg(swift_libraries['SWIFT_LIB_STATIC'])
+        ctx.env['SWIFT_LIB_STATIC'] = swift_libraries['SWIFT_LIB_STATIC']
+    else:
+        ctx.end_msg(False)
+
+    enableStatic = getattr(ctx.options, 'enable_swift-static')
+    if (enableStatic) and 'SWIFT_LIB_STATIC' in swift_libraries:
+        __add_static_swift_library_linking_flags(ctx, swift_libraries['SWIFT_LIB_STATIC'])
+    else:
+        __add_dynamic_swift_library_linking_flags(ctx, swift_libraries['SWIFT_LIB_DYNAMIC'])
 
 def __find_macos_sdk(ctx):
     ctx.start_msg('Checking for macOS SDK')
-    sdk = __run(['xcrun', '--sdk', 'macosx', '--show-sdk-path'])
+    sdk = None
+    sdk_build_version = None
+    sdk_version = None
+
+    #look for set macOS SDK paths and version in passed environment variables
+    if 'MACOS_SDK' in ctx.environ:
+        sdk = ctx.environ['MACOS_SDK']
+    if 'MACOS_SDK_VERSION' in ctx.environ:
+        ctx.env.MACOS_SDK_VERSION = ctx.environ['MACOS_SDK_VERSION']
+
+    #find macOS SDK paths and version
+    if not sdk:
+        sdk = __run(['xcrun', '--sdk', 'macosx', '--show-sdk-path'])
+    if not ctx.env.MACOS_SDK_VERSION:
+        #show-sdk-build-version: is not available on older command line tools, but return a build version (eg 17A360)
+        #show-sdk-version: is always available, but on older dev tools it's only the major version
+        sdk_build_version = __run(['xcrun', '--sdk', 'macosx', '--show-sdk-build-version' ])
+        sdk_version = __run(['xcrun', '--sdk', 'macosx', '--show-sdk-version' ])
+
     if sdk:
-        ctx.end_msg(sdk)
         ctx.env.MACOS_SDK = sdk
+        build_version = '10.10.0'
+
+        if not ctx.env.MACOS_SDK_VERSION:
+            #convert build version to a version string
+            #first 2 two digits are the major version, starting with 15 which is 10.11 (offset of 4)
+            #1 char is the minor version, A => 0, B => 1 and ongoing
+            #las digits are bugfix version, which are nor relevant for us
+            #eg 16E185 => 10.12.4, 17A360 => 10.13, 18B71 => 10.14.1
+            if sdk_build_version and isinstance(sdk_build_version, str):
+                verRe = re.compile("(\d+)(\D+)(\d+)")
+                version_parts = verRe.search(sdk_build_version)
+                major = int(version_parts.group(1))-4
+                minor = string.ascii_lowercase.index(version_parts.group(2).lower())
+                build_version = '10.' + str(major) + '.' + str(minor)
```

Comment:
I don't have a heavy opinion in this case regarding which is better, but just noting that there's the `"10.{}.{}".format(major, minor)` way of formatting strings as well :) .